general/development/policies/codingstyle/frankenstyle.md

```diff
@@ -0,0 +1,173 @@
+| Grades | core_grades | /grade |
+| Advanced grading | core_grading | /grade/grading |
+| Groups | core_group | /group |
+| Help | core_help |  |
+| Hub | core_hub |  |
+| IMS CC | core_imscc |  |
+| Installer | core_install |  |
+| ISO 6392 | core_iso6392 |  |
+| Language pack configuration | core_langconfig |  |
+| License | core_license |  |
+| Maths library | core_mathslib |  |
+| Media | core_media |  |
+| Messaging | core_message | /message |
+| MIME types | core_mimetypes |
+| MNet | core_mnet | /mnet |
+| Dashboard | core_my | /my |
+| User notes | core_notes | /notes |
+| Page types | core_pagetype |  |
+| Pictures and icons | core_pix |  |
+| Plagiarism | core_plagiarism | /plagiarism |
+| Plugins management | core_plugin |  |
+| Portfolio | core_portfolio | /portfolio |
+| Privacy | core_privacy | /privacy |
+| Course publishing | core_publish | /course/publish |
+| Question bank engine | core_question | /question |
+| Ratings | core_rating | /rating |
+| Site registration | core_register | /admin/registration |
+| Repository | core_repository | /repository |
+| RSS | core_rss | /rss |
+| Roles | core_role | /admin/roles |
+| Global search | core_search | /search |
+| Tabular data display/download (deprecated) | core_table |  |
+| Tagging | core_tag | /tag |
+| Timezones | core_timezones |  |
+| User | core_user | /user |
+| User key | core_userkey |  |
+| Web service | core_webservice | /webservice |
+
+## Usage
+
+Frankenstyle component names are used in:
+
+### Function names
+
+All plugin functions must start with full frankenstyle prefix. For backwards compatibility modules may also use **modulename_** as prefix.
+
+### Class names
+
+- All the component classes must sit under the classes ([auto-loaded](https://docs.moodle.org/dev/Automatic_class_loading)) directory, belonging to their frankenstyle namespace and having a natural name, ex: \mod_forum\example.
+- The use of class names using the frankenstyle prefix is now deprecated (see [Coding style](./index.md#namespaces)), ex: mod_forum_example.
+
+### Constants
+
+All plugin constants must start with uppercase frankenstyle prefix, example MOD_FORUM_XXXX.
+
+### Table names
+
+All table names for a plugin must begin with its frankenstyle name (after the standard Moodle table prefix).  
+
+(The exception to this rule is Moodle activities which (for historical reasons) do not have mod_ in front of the plugin name)
+
+Examples: mdl\_**local_`coolreport`**, mdl\_**local_`coolreport`**_users
+
+### Plugin configuration table
+
+In the **config_plugins** table, column **plugin**, the frankenstyle name is used.
+
+### Capabilities
+
+All capabilities for a plugin use the frankenstyle name, except with a / instead of a _.
+
+Example: **mod/quiz**:`viewattempt`
+
+### Language files
+
+The main language file for each plugin (with the notable exception of activity modules) is the frankenstyle component name.
+
+Examples:  /blocks/participants/lang/en/**block_participants**.php
```

Comment:
You get the drill - try to give multiple examples and includea ny exceptions (liek activity modules in this case).

Again, the exceptions (mod_) should be in an admonition like a `:::warning`